java/src/main/java/org/rocksdb/SstFileMetaData.java

```diff
@@ -8,57 +8,17 @@
 /**
  * The metadata that describes a SST file.
  */
-public class SstFileMetaData {
-  private final String fileName;
-  private final String path;
-  private final long size;
-  private final long smallestSeqno;
-  private final long largestSeqno;
-  private final byte[] smallestKey;
-  private final byte[] largestKey;
-  private final long numReadsSampled;
-  private final boolean beingCompacted;
-  private final long numEntries;
-  private final long numDeletions;
-
+public class SstFileMetaData extends RocksObject {
   /**
-   * Called from JNI C++
-   *
-   * @param fileName the file name
-   * @param path the file path
-   * @param size the size of the file
-   * @param smallestSeqno the smallest sequence number
-   * @param largestSeqno the largest sequence number
-   * @param smallestKey the smallest key
-   * @param largestKey the largest key
-   * @param numReadsSampled the number of reads sampled
-   * @param beingCompacted true if the file is being compacted, false otherwise
-   * @param numEntries the number of entries
-   * @param numDeletions the number of deletions
+   * Called from JNI C++ or subclass
+   * @param nativeHandle pointer to the C++ object
    */
-  protected SstFileMetaData(
-      final String fileName,
-      final String path,
-      final long size,
-      final long smallestSeqno,
-      final long largestSeqno,
-      final byte[] smallestKey,
-      final byte[] largestKey,
-      final long numReadsSampled,
-      final boolean beingCompacted,
-      final long numEntries,
-      final long numDeletions) {
-    this.fileName = fileName;
-    this.path = path;
-    this.size = size;
-    this.smallestSeqno = smallestSeqno;
-    this.largestSeqno = largestSeqno;
-    this.smallestKey = smallestKey;
-    this.largestKey = largestKey;
-    this.numReadsSampled = numReadsSampled;
-    this.beingCompacted = beingCompacted;
-    this.numEntries = numEntries;
-    this.numDeletions = numDeletions;
+  public SstFileMetaData(final long nativeHandle) {
```

Comment:
So that we can convert `LiveFileMetaData` to it's parent class `SstFileMetaData`.